source/cipher.c

```diff
@@ -376,8 +380,14 @@ int aws_cryptosdk_encrypt_body(
          * just in case.
          */
         if (!aws_byte_cursor_advance_nospec(&incurs, in_len).ptr) goto out;
+#ifdef CBMC
+#    pragma CPROVER check push
+#    pragma CPROVER check disable "conversion"
+#endif
         outbuf.len += ct_len;
```

Comment:
Is this guaranteed never to fail? Why?

Maybe it would be nice to put a dynamic check that it doesn't overflow, in the code.